src/Model/Item.php

```diff
@@ -895,10 +895,14 @@ public static function update(array $fields, array $condition)
 				}
 			}
 
-			if (!empty($tags)) {
-				Term::insertFromTagFieldByItemId($item['id'], $tags);
-				if (!empty($item['tag'])) {
-					DBA::update('item', ['tag' => ''], ['id' => $item['id']]);
+			if (!is_null($tags)) {
```

Comment:
How do you decide whether the tags should be emptied or if the update doesn't contain the tags at all? Example: When you only update the "body", this part of the code mustn't be executed.

Reply:
I didn't think "*insert*FromTagFieldByItemId" was doing a deletion as well, so you actually can replace 
```php
				if ($tags) {
					Term::insertFromTagFieldByItemId($item['id'], $tags);
					if (!empty($item['tag'])) {
						DBA::update('item', ['tag' => ''], ['id' => $item['id']]);
					}
				}
				else {
					Term::deleteAllTags($item['id']);
				}
```
by
```php
				Term::insertFromTagFieldByItemId($item['id'], $tags);
				if (!empty($item['tag'])) {
					DBA::update('item', ['tag' => ''], ['id' => $item['id']]);
				}
```
because if `$tags` is an empty string tags will be deleted and none will be added in `Term::insertFromTagFieldByItemId`.

Reply:
I'm a bit confused where the deletion should happen than. Should `tagrm` check if the array with the tags is empty and delete the old one if necessary ?

Reply:
When the "tag" field isn't present in the current update command, we mustn't invoke the tag functions. if we wouldn't do so, an update to the "title" field (without having the "tag" in the field list) would remove all tags from the item.